### ISSUE TITLE
Gridmaps should preview their octant groupings in the editor

### ISSUE THREAD

**Issue by jamie-pate** [godotengine/godot-proposals#1824]
**Describe the project you are working on:**
An office layout that uses gridmaps to draw the walls, floors and ceilings

**Describe the problem or limitation you are having in your project:**
Trying to light the space is a nightmare because the octant sizing is not clear.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Toggle or permanently show a gizmo that visualizes the octant size as it's calculated based on the cell size and octant size settings. This would allow designers to easily visualize how many lights may be placed in one area, and how the engine is grouping the tiles they placed.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
The Editor would gain a new gizmo 'Gridmap Octants' which can be shown/hidden based on gridmap selection etc. The gizmo would draw cubes for each octant so that designers can easily see the boundaries and the effect of changing the cell size and octant size.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
It possibly could be implemented as a plugin which adds a new gizmo.

**Is there a reason why this should be core and not an add-on in the asset library?:**
It would be useful for anyone using GridMaps, especially with the current 3.2 limitation on the number of lights allowed to affect the single MultiMeshInstance that's created for each octant. This lighting restriction is supposed to be removed in the Vulcan renderer, but will also it be removed from the OpenGL renderer for 4.0? It'd also be a great way to visualize the complexity of the gridmap because it's currently difficult to guess what happens when you change cell size and octant size without reading the engine source code.

**Comment by Calinou** [comment]
> This lighting restriction is supposed to be removed in the Vulcan renderer, but will also it be removed from the OpenGL renderer for 4.0?

Yes, as any OpenGL renderer that will be implemented will use multi-pass lighting to make the renderer more compatible. This is slower, but it won't have a limitation in the number of lights you can use.

**Comment by Calinou** [comment]
@jamie-pate Are you still running into limitations with light count when using GridMaps? This should be much less frequent in 3.3 and later, now that you can use 32 lights per mesh resource (by default) in GLES3.

The 8-lights-per-mesh-resource limitation is still present in the Vulkan Mobile backend, but given that backend's focus on low-power devices, it may not be a good idea to use a lot of lights with GridMap there.

**Comment by jamie-pate** [comment]
3.3 resolved my headroom issues with the number of lights.
